Movie Search/MovieSearchViewController.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  MovieSearchViewController.swift
+//  Movie Search
+//
+//  Created by Alexey Grabik on 15/04/2020.
+//
+
+import UIKit
+
+class MovieSearchViewController: UIViewController {
+
+    @IBOutlet var tableView: UITableView!
+    @IBOutlet var statusLabel: UILabel!
+    @IBOutlet var searchBar: UISearchBar!
+    @IBOutlet var spinner: UIActivityIndicatorView!
+    let searchController = SearchController()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupSearchCallback()
+        setupTable()
+        setInitialStatus()
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        searchBar.becomeFirstResponder()
+    }
+}
+
+private extension MovieSearchViewController {
+    
+    func setupSearchCallback() {
+        searchController.searchPerformedCallback = { [weak self] status in
+            guard let self = self else { return }
```

Comment:
is this callback called on the main thread?

Reply:
yes